cg/meta/create_validation_cases/validation_cases_api.py

```diff
@@ -0,0 +1,187 @@
+        """Return the status of a sample."""
+        return sample.links[0].status if sample.links else "unknown"
+
+    def create_validation_case_in_statusdb(self, validation_case_data: ValidationCaseData) -> None:
+        """
+        Add the validation samples and case to statusDB and generate the sample case link.
+        """
+        try:
+            self.store_validation_samples(validation_case_data)
+            self.store_validation_case(validation_case_data)
+            self._link_sample_to_case(validation_case_data)
+            self.status_db.session.commit()
+        except Exception as error:
+            LOG.warning(f"An error occurred {repr(error)}")
+            self.status_db.session.rollback()
+
+    def create_validation_sample_bundles(self, validation_samples: list[Sample]) -> None:
+        for validation_sample in validation_samples:
+            self.hk_api.create_new_bundle_and_version(validation_sample.internal_id)
+
+    def copy_original_sample_bundle_files(self, validation_samples: list[Sample]) -> None:
+        """Copy the fastq files from the original sample bundle into the new validation sample bundle"""
+        for validation_sample in validation_samples:
+            fastq_files: list[File] = self.hk_api.get_files_from_latest_version(
+                bundle_name=validation_sample.from_sample, tags=[SequencingFileTag.FASTQ]
+            )
+            validation_bundle_path: Path = self.hk_api.get_latest_bundle_version(
+                bundle_name=validation_sample.internal_id
+            ).full_path
+            for fastq_file in fastq_files:
+                copy_file(file_path=Path(fastq_file.full_path), destination=validation_bundle_path)
+
+    def rename_and_add_original_sample_bundle_files(self, validation_samples: list[Sample]) -> None:
+        """Rename the fastq files to use the new validation sample internal id."""
+        for validation_sample in validation_samples:
+            validation_bundle_path: Path = self.hk_api.get_latest_bundle_version(
+                bundle_name=validation_sample.internal_id
+            ).full_path
+            fastq_files: list[Path] = get_files_matching_pattern(
+                directory=validation_bundle_path, pattern=SequencingFileTag.FASTQ
+            )
+            for fastq_file in fastq_files:
+                new_fast_file_path = Path(
+                    fastq_file.__str__().replace(validation_sample.from_sample),
+                    validation_sample.internal_id,
+                )
+                rename_file(file_path=fastq_file, renamed_file_path=new_fast_file_path)
+                new_tags: list[str] = self.get_new_tags(validation_sample)
+                self.add_validation_samples_to_hk(
+                    sample_id=validation_sample.internal_id,
+                    file_path=new_fast_file_path,
+                    tags=new_tags,
+                )
+
+    def get_new_tags(self, validation_sample: Sample) -> list[str]:
```

Comment:
what if the case is compressed and we do not have the fastq tag?

Reply:
You will need to make sure the case is available and decompressed at the moment.